Site: Document createEventDispatcher.

pngwn opened the pull request:
Documents `createEventDispatcher`. I don't know if all of this information should be here but the `on` directive is pretty sparse and just links to this part of the docs, so I included information about listening and the `payload` as well.

I was also unsure about the function signature for this, I couldn't see anything quite like it.

#2533

pngwn commented:
I've made this change now. The Component Event documentation seems incomplete to me, however, it basically just links down to the `createEventDispatcher` docs.

pngwn commented:
Additionally, should `payload` become `detail`? That is more consistent with the internal naming and the property the payload appears on. I used `payload` because I felt it is a more universal term but consistency might be preferable here.

mindrones commented:
> The Component Event documentation seems incomplete to me

The last phrase ends with A ":" indeed so it would be nice to fix that at least and add what you feel it's misssing.

> should `payload` become `detail`? 

I too prefer `payload` but then it's counter-intuitive having the payload on `detail`. It might be worth renaming, but I'll let it to @Rich-Harris ofc.

pngwn commented:
I've have added a mention of forwarding component events in the 'Template Syntax' > 'Component Events' section, even though they behave the same as DOM events, I felt it was worth mentioning for clarity.

I've rewritten the `createEventDispatcher` documentation to explain that it is a `CustomEvent` and added information about bubbling, cancelling and the detail property. All mentions of `payload` have been changed to `detail` and `CustomEvent.detail`. 

I have included links to MDN where relevant.

mindrones commented:
As someone once said: "good job, person who wrote the docs"!
